Form/AbstractModifySettingsForm.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Craue\ConfigBundle\Form;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\HttpKernel\Kernel;
+
+/**
+ * @author Christian Raue <christian.raue@gmail.com>
+ * @copyright 2011-2015 Christian Raue
+ * @license http://opensource.org/licenses/mit-license.php MIT License
+ */
+abstract class AbstractModifySettingsForm extends AbstractType {
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function buildForm(FormBuilderInterface $builder, array $options) {
+		$builder->add('settings', Kernel::VERSION_ID < 20800 ? 'collection' : 'Symfony\Component\Form\Extension\Core\Type\CollectionType', array(
+			'type' => Kernel::VERSION_ID < 20800 ? 'craue_config_setting' : 'Craue\ConfigBundle\Form\Type\SettingType',
```

Comment:
```php
if (method_exists(AbstractType::class, 'getBlockPrefix')) {
    // Form component 2.8+
} else {
    // BC with Form 2.7 and older
}
```

note that my snippet use the PHP 5.5+ feature, but you can also put the class name in a string directly for older version support